### ISSUE TITLE
(Production & Staging) "stop" should pause Spotify

### ISSUE THREAD

**Issue by tonyespinoza1** [stanford-oval/thingpedia-common-devices#382]
"stop" should pause playback (when started by Genie).  no other response expected.

(logs below are from Production but I verified the issue exists on Staging as well)

Hello! How can I help you?
play music

What song do you want to play?
beatles

I played music by The Beatles on Spotify.
another song

Sorry, I did not understand that. Can you rephrase it?
play another song

I played the next song.
stop
stop
please stop

Alright, let me know if I can help you with anything else!

```
====
# web-fa81bab2
#! timestamp: 2021-07-18T00:31:09.445Z
U: play music
UT: @com.spotify.play();
C: $dialogue @org.thingpedia.dialogue.transaction.execute;
C: @com.spotify(id="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza")).play();
#! timestamp: 2021-07-18T00:31:10.323Z
A: What song do you want to play?
AT: $dialogue @org.thingpedia.dialogue.transaction.sys_slot_fill(playable);
AT: @com.spotify(id="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza")).play();
#! timestamp: 2021-07-18T00:31:19.914Z
U: beatles
UT: $dialogue @org.thingpedia.dialogue.transaction.execute;
UT: @com.spotify(id="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza")).playable() filter contains(artists, null^^com.spotify:artist("beatles")) => @com.spotify.play(playable=id);
C: $dialogue @org.thingpedia.dialogue.transaction.execute;
C: @com.spotify(id="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza")).playable() filter contains(artists, "spotify:artist:3WrFJ7ztbogyGnTHbHJFl2"^^com.spotify:artist("The Beatles")) => @com.spotify(id="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza")).play(playable=id)
C: #[results=[
C:   { value="2faa93e1b2b45b85b02b90b497d311c37e7659e7", display="Soundworks’s Mac Pro", playable="spotify:track:6dGnYIeXmHdcikdzNNDMm2"^^com.spotify:playable("Here Comes The Sun - Remastered 2009"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { value="2faa93e1b2b45b85b02b90b497d311c37e7659e7", display="Soundworks’s Mac Pro", playable="spotify:track:2EqlS6tkEnglzr7tkKAAYD"^^com.spotify:playable("Come Together - Remastered 2009"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { value="2faa93e1b2b45b85b02b90b497d311c37e7659e7", display="Soundworks’s Mac Pro", playable="spotify:track:7iN1s7xHE4ifF5povM6A48"^^com.spotify:playable("Let It Be - Remastered 2009"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { value="2faa93e1b2b45b85b02b90b497d311c37e7659e7", display="Soundworks’s Mac Pro", playable="spotify:track:3BQHpFgAp4l80e1XslIjNI"^^com.spotify:playable("Yesterday - Remastered 2009"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { value="2faa93e1b2b45b85b02b90b497d311c37e7659e7", display="Soundworks’s Mac Pro", playable="spotify:track:5jgFfDIR6FR0gvlA56Nakr"^^com.spotify:playable("Blackbird - Remastered 2009"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { value="2faa93e1b2b45b85b02b90b497d311c37e7659e7", display="Soundworks’s Mac Pro", playable="spotify:track:5ZBeML7Lf3FMEVviTyvi8l"^^com.spotify:playable("Twist And Shout - Remastered 2009"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { value="2faa93e1b2b45b85b02b90b497d311c37e7659e7", display="Soundworks’s Mac Pro", playable="spotify:track:0aym2LBJBk9DAYuHHutrIl"^^com.spotify:playable("Hey Jude - Remastered 2015"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { value="2faa93e1b2b45b85b02b90b497d311c37e7659e7", display="Soundworks’s Mac Pro", playable="spotify:track:3KfbEIOC7YIv90FIfNSZpo"^^com.spotify:playable("In My Life - Remastered 2009"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { value="2faa93e1b2b45b85b02b90b497d311c37e7659e7", display="Soundworks’s Mac Pro", playable="spotify:track:4pbG9SUmWIvsROVLF0zF9s"^^com.spotify:playable("I Want To Hold Your Hand - Remastered 2015"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { value="2faa93e1b2b45b85b02b90b497d311c37e7659e7", display="Soundworks’s Mac Pro", playable="spotify:track:3Am0IbOxmvlSXro7N5iSfZ"^^com.spotify:playable("Strawberry Fields Forever - Remastered 2009"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") }
C: ]]
C: #[count=47];
#! timestamp: 2021-07-18T00:31:21.085Z
A: I played music by The Beatles on Spotify.
AT: $dialogue @org.thingpedia.dialogue.transaction.sys_action_success;
#! vote: up
#! timestamp: 2021-07-18T00:31:33.795Z
U: another song
UT: $answer(new Location("another song"));
#! timestamp: 2021-07-18T00:31:33.795Z
A: Sorry, I did not understand that. Can you rephrase it?
====
# web-fa81bab2
#! timestamp: 2021-07-18T00:31:37.369Z
U: play another song
UT: $dialogue @org.thingpedia.dialogue.transaction.execute;
UT: @com.spotify.player_next();
C: $dialogue @org.thingpedia.dialogue.transaction.execute;
C: @com.spotify(id="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza")).player_next()
C: #[results=[]];
#! timestamp: 2021-07-18T00:31:37.827Z
A: I played the next song.
AT: $dialogue @org.thingpedia.dialogue.transaction.sys_action_success;
#! vote: up
#! timestamp: 2021-07-18T00:31:49.105Z
U: stop
UT: $stop;
====
# web-fa81bab2
#! timestamp: 2021-07-18T00:32:03.230Z
U: stop
UT: $stop;
====
# web-fa81bab2
#! timestamp: 2021-07-18T00:32:08.706Z
U: please stop
UT: $dialogue @org.thingpedia.dialogue.transaction.cancel;
C: $dialogue @org.thingpedia.dialogue.transaction.cancel;
#! timestamp: 2021-07-18T00:32:08.967Z
A: Alright, let me know if I can help you with anything else!
AT: $dialogue @org.thingpedia.dialogue.transaction.sys_end;
```

**Comment by gcampax** [comment]
I think this is an issue with the Spotify skill not registering with the audio controller correctly.

**Comment by tonyespinoza1** [comment]
Here's another case where it failed to respond to "stop" 

```
====
# main/fe5cce10-1024-42f4-b290-31cf8da10b43
#! timestamp: 2021-07-31T01:20:27.755Z
U: play spotify
UT: $dialogue @org.thingpedia.dialogue.transaction.execute;
UT: @org.thingpedia.media-source.playable() => @org.thingpedia.media-player.play(playable=id);
C: $dialogue @org.thingpedia.dialogue.transaction.execute;
C: @org.thingpedia.media-source(id="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza")).playable() => @org.thingpedia.media-player(id="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza")).play(playable=id)
C: #[results=[
C:   { playable="spotify:track:4ZtFanR9U6ndgddUvNcjcG"^^org.thingpedia.media-source:playable("good 4 u"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { playable="spotify:track:0UsmyJDsst2xhX1ZiFF3JW"^^org.thingpedia.media-source:playable("Year,2015"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { playable="spotify:track:7BFk1nSfwfkDOgnNNeY7Yn"^^org.thingpedia.media-source:playable("wish i dropped out like brakence interlude"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { playable="spotify:track:0eRJI4VhEm6UqIdMbf4VyX"^^org.thingpedia.media-source:playable("Year 3000"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { playable="spotify:track:5HCyWlXZPP0y6Gqq8TgA20"^^org.thingpedia.media-source:playable("STAY (with Justin Bieber)"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { playable="spotify:track:5rJw9VsPNdfnV9Ar97xZG2"^^org.thingpedia.media-source:playable("K-LOVE Fan Awards: Songs of the Year (2015 Mash-Up)"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { playable="spotify:track:1lrjBRhWKrd3YShBN4Un7c"^^org.thingpedia.media-source:playable("Bob Dylan"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { playable="spotify:track:5BJs3aDrKYBRkzrhaVxf89"^^org.thingpedia.media-source:playable("After Me - 2021 Version"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { playable="spotify:track:748mdHapucXQri7IAO8yFK"^^org.thingpedia.media-source:playable("Kiss Me More (feat. SZA)"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { playable="spotify:track:00ohIpPn9LkKpeIqhfIU9V"^^org.thingpedia.media-source:playable("K-Love Fan Awards: Songs of the Year (2014 Mash-Up)"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") }
C: ]]
C: #[count=50];
#! timestamp: 2021-07-31T01:20:29.804Z
A: I played music on Spotify.
AT: $dialogue @org.thingpedia.dialogue.transaction.sys_action_success;
#! timestamp: 2021-07-31T01:20:37.670Z
U: add song to oval
UT: $dialogue @org.thingpedia.dialogue.transaction.execute;
UT: @org.thingpedia.media-source(id="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza")).song() => @com.spotify.add_song_to_playlist(playlist="oval", song=id);
C: $dialogue @org.thingpedia.dialogue.transaction.execute;
C: @org.thingpedia.media-source(id="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza")).playable() => @org.thingpedia.media-player(id="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza")).play(playable=id)
C: #[results=[
C:   { playable="spotify:track:4ZtFanR9U6ndgddUvNcjcG"^^org.thingpedia.media-source:playable("good 4 u"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { playable="spotify:track:0UsmyJDsst2xhX1ZiFF3JW"^^org.thingpedia.media-source:playable("Year,2015"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { playable="spotify:track:7BFk1nSfwfkDOgnNNeY7Yn"^^org.thingpedia.media-source:playable("wish i dropped out like brakence interlude"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { playable="spotify:track:0eRJI4VhEm6UqIdMbf4VyX"^^org.thingpedia.media-source:playable("Year 3000"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { playable="spotify:track:5HCyWlXZPP0y6Gqq8TgA20"^^org.thingpedia.media-source:playable("STAY (with Justin Bieber)"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { playable="spotify:track:5rJw9VsPNdfnV9Ar97xZG2"^^org.thingpedia.media-source:playable("K-LOVE Fan Awards: Songs of the Year (2015 Mash-Up)"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { playable="spotify:track:1lrjBRhWKrd3YShBN4Un7c"^^org.thingpedia.media-source:playable("Bob Dylan"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { playable="spotify:track:5BJs3aDrKYBRkzrhaVxf89"^^org.thingpedia.media-source:playable("After Me - 2021 Version"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { playable="spotify:track:748mdHapucXQri7IAO8yFK"^^org.thingpedia.media-source:playable("Kiss Me More (feat. SZA)"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { playable="spotify:track:00ohIpPn9LkKpeIqhfIU9V"^^org.thingpedia.media-source:playable("K-Love Fan Awards: Songs of the Year (2014 Mash-Up)"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") }
C: ]]
C: #[count=50];
C: @org.thingpedia.media-source(id="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza")).song() => @com.spotify(id="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza")).add_song_to_playlist(playlist="oval", song=id)
C: #[results=[
C:   { playlist="oval", song="spotify:track:4ZtFanR9U6ndgddUvNcjcG"^^org.thingpedia.media-source:song("good 4 u"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { playlist="oval", song="spotify:track:67BtfxlNbhBmCDR2L2l8qd"^^org.thingpedia.media-source:song("MONTERO (Call Me By Your Name)"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { playlist="oval", song="spotify:track:5HCyWlXZPP0y6Gqq8TgA20"^^org.thingpedia.media-source:song("STAY (with Justin Bieber)"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { playlist="oval", song="spotify:track:6PQ88X9TkUIAUIZJHW2upE"^^org.thingpedia.media-source:song("Bad Habits"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { playlist="oval", song="spotify:track:2JPLbjOn0wPCngEot2STUS"^^org.thingpedia.media-source:song("Yonaguni"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { playlist="oval", song="spotify:track:2bgTY4UwhfBYhGT4HUYStN"^^org.thingpedia.media-source:song("Butter"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { playlist="oval", song="spotify:track:0LThjFY2iTtNdd4wviwVV2"^^org.thingpedia.media-source:song("Permission to Dance"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { playlist="oval", song="spotify:track:748mdHapucXQri7IAO8yFK"^^org.thingpedia.media-source:song("Kiss Me More (feat. SZA)"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { playlist="oval", song="spotify:track:4iJyoBOLtHqaGxP12qzhQI"^^org.thingpedia.media-source:song("Peaches (feat. Daniel Caesar & Giveon)"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") },
C:   { playlist="oval", song="spotify:track:5CZ40GBx1sQ9agT82CLQCT"^^org.thingpedia.media-source:song("traitor"), __device="com.spotify-126743886"^^tt:device_id("Spotify Account of Tony Espinoza") }
C: ]]
C: #[count=50];
#! timestamp: 2021-07-31T01:21:01.095Z
A: Did you mean Get songs and then add a song to a playlist with name oval and name the id?
A: I added a song to a playlist.
AT: $dialogue @org.thingpedia.dialogue.transaction.sys_action_success;
#! timestamp: 2021-07-31T01:24:54.968Z
U: stop
UT: $stop;

```

**Comment by tonyespinoza1** [comment]
ideally "stop" would be a global command that applies to any skill that is streaming audio

**Comment by gcampax** [comment]
That is in fact the design, so there is a bug somewhere in the implementation...

**Comment by ryachen01** [comment]
This problem doesn't happen when running a local instance, so I think the issue here is that "stop" calls the stopAudio function of the audio controller, but we only set a release callback in Spotify if we're running spotifyd. @gcampax You mentioned earlier that the audio controller should only be used with spotifyd. How should we handle this?

**Comment by gcampax** [comment]
We should unconditionally register the skill with the audio controller. We need this functionality in the cloud as well.

**Comment by tonyespinoza1** [comment]
I want to begin testing the system with my 4 year old daughter ASAP and I think it will be the most important thing.  Will basically build a set of spotify playlists she can control...  Not expecting a fine grain of control for her, but STOP will be key.

**Comment by gcampax** [comment]
"stop" works when playing from a Genie client. It won't stop music playing from a Spotify app (use "stop the music" for that). That's by design because "stop" should only affect the assistant. Closing.
